### PR TITLE
Check for irrevocable IO in AdvTester sink; warn otherwise

### PR DESCRIPTION
**Pull request by albert-magyar**
This now has a TestErrorLog (derivative of chisel3.internal.ErrorLog) to allow the tester to produce warnings in a style matching Chisel3 errors/infos/warnings.

### CODE REVIEW

File: src/main/scala/chisel3/iotesters/AdvTester.scala [hunk 1]
```diff
@@ -98,9 +101,19 @@ abstract class AdvTester[+T <: Module](dut: T,
     expr
   }
 
-  class DecoupledSink[T <: Data, R]( socket: DecoupledIO[T], cvt: T=>R, 
+  class IrrevocableSink[T <: Data, R]( socket: ReadyValidIO[T], cvt: T=>R,
```
Comment (chick) [inline src/main/scala/chisel3/iotesters/AdvTester.scala]: Shouldn't  DecoupledSink be kept around with a deprecation instead.  There must be users who will be have change there code if this change goes through

Comment (albert-magyar) [inline src/main/scala/chisel3/iotesters/AdvTester.scala]: That sounds like a good idea since I already added the warnings.

Comment (albert-magyar) [inline src/main/scala/chisel3/iotesters/AdvTester.scala]: I addressed this by making DecoupledSink a subclass of IrrevocableSink

**Comment by albert-magyar** [comment]
@chick, I can't merge this since I don't have write access.

**Pull request merged**